eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -194,6 +195,63 @@ default T getOnly()
         return this.anySatisfy(each -> Objects.equals(value, function.valueOf(each)));
     }
 
+    /**
+     * Returns true if any of the elements in source is contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(Collection<T> source)
+    {
+        Objects.requireNonNull(source, "source Collection can't be null");
+        for (T object : source)
+        {
+            if (this.contains(object))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if none of the elements in source are contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsNone(Collection<T> source)
+    {
+        return !this.containsAny(source);
+    }
+
+    /**
+     * Returns true if any of the elements in source is contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsAnyIterable(Iterable<T> source)
+    {
+        Objects.requireNonNull(source, "source Iterable can't be null");
+        Iterator<T> iterator = source.iterator();
```

Comment:
Greetings @donraab! So I've to implement these methods in both `RichIterable` and `AbstractRichIterable`?

Reply:
So, I've tried to do the latter and realised it doesn't work that way and will result in errors (I would have to implement those methods in all the subclasses which is tedious 😄 ). I've written them as default methods in `RichIterable` by checking if the input is an `instanceof` `RichIterable` then calling `anySatisfy` and if it's not I used `StreamSupport.stream` to convert the input into a stream and call `anyMatch` on it. Let me know your thoughts on this... 👍

Reply:
Hi @hvydya, thanks for pushing the updates. I will look today. I understand and certainly appreciate not using default methods may be tedious. My initial gut feeling is to not use Streams in the default methods. But I'll see if there is a better alternative.

Reply:
Hi @hvydya, I looked at the implementation, and it is simple and clean. The upside of using `Stream` is that it gives you access to use `anyMatch` and `noneMatch`. The downside of using `Stream` is that it creates potentially unnecessary garbage. With JDK Collections, the collections that we can avoid creating an `Iterator` or `Stream` are `ArrayList` and any `RandomAccess` `List`.

However, this is not the biggest issue in terms of performance. There are potentially performance optimizations that have been used for `containsAll`, `containsAny` and `containsNone` on the primitive collections that we could leverage on the object side.

Look at the following PR and see the approach that we are taking to see how we could leverage the same for `containsAny` and containsNone.

https://github.com/eclipse/eclipse-collections/pull/1016

Reply:
Hi @donraab , I looked at the pr you suggested. This is what I tested and came up with:
```
    default boolean containsAny(Collection<?> source)
    {
        Iterator<?> iter = source.iterator();
        Set<T> current = this.toSet();
        while (iter.hasNext()) {
            if (current.contains(iter.next())) {
                return true;
            }
        }
        return false;
    }
```
I am not sure if this is the best way to write it. Let me know your thoughts on this...

Reply:
The more I look at the code more I think there is no other way but to implement in each of the individual classes without giving a default implementation. Or the default implementation can just throw an exception for backwards compatibility

Reply:
Individual classes as in the implementations of `RichIterable` ? If that's what you meant then there are 439 implementations according to my IDE (IntelliJ). I am interested to make the change if that's the best way to do this.

Reply:
I’ll let @donraab make the decision since he is close to the change than I am. 
As far as the 459 implementations go, most of them will have abstract classes so it’ll end up being far less. Some of them will be on the primitive collections so they will be code generated.

Reply:
Alright @nikhilnanivadekar . Waiting for your comments @donraab ...

Reply:
@nikhilnanivadekar @hvydya Apologies, looks like I missed following up on this comment thread. 

I agree with Nikhil that this is going to have to be implemented in each of the individual classes. The primitive implementation of containsAny and containsNone that was worked on by @rinatigati should be used as a guide in terms of implementation for any performance optimizations.